template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ConfigUtils.java

```diff
@@ -0,0 +1,30 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders;
+
+import java.util.Optional;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.base.DatabaseType;
+import com.sequenceiq.cloudbreak.domain.RDSConfig;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+
+public class ConfigUtils {
+
+    private ConfigUtils() { }
+
+    /**
+     * Convenience method for creating an {@link ApiClusterTemplateConfig} with a value.
+     *
+     * @param name config name
+     * @param value config value
+     * @return config object
+     */
+    public static ApiClusterTemplateConfig config(String name, String value) {
+        return new ApiClusterTemplateConfig().name(name).value(value);
+    }
+
+    public static Optional<RDSConfig> getFirstRDSConfigOptional(TemplatePreparationObject source, DatabaseType databaseType) {
```

Comment:
Picking the first DB does not feel correct. I think if there are more than one from the same type e.g two ranger DBs then it should be an error case.

I also think it would be great to investigate whether we need less than 1 is acceptable or is an error path. E.g in case of ranger if you have no database then definitely it is an error.

Reply:
Good points, but I have not changed this behavior, just extracted common logic (from Hive and Oozie config providers).  It is not used for Ranger.

Reply:
Understood. Then I think we can resolve this, but would you be so kind as to raise a jira to follow up on this? And if you have time please fix it in your next pull request, or if you don't have time could you assign that Jira to the persons who has implemented hive, oozie, etc. please? You can use git blame :)

Reply:
Opened a ticket.